md-rest-api/src/main/java/com/wipro/ats/bdre/md/rest/AppDeploymentQueueAPI.java

```diff
@@ -46,9 +42,182 @@
+                    tableProperties.setKey(daoProperties.getId().getPropKey());
+                    tableProperties.setValue(daoProperties.getPropValue());
+                    tableProperties.setDescription(daoProperties.getDescription());
+                    propertiesList.add(tableProperties);
+                }}
+            processExport.setProcessList(processList);
+            processExport.setPropertiesList(propertiesList);
+            LOGGER.info("export object is "+processExport);
+            AddJson addJson=new AddJson();
+            String status=addJson.addJsonToProcessId(appDeploymentQueue.getProcessId().toString(),processExport);
+            LOGGER.info("status of process.json addition "+status);
+            com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueue jpaAppDeploymentQueue=new com.wipro.ats.bdre.md.dao.jpa.AppDeploymentQueue();
+            jpaAppDeploymentQueue.setAppDeploymentQueueStatus(appDeploymentQueueStatusDAO.get((short) 0));
+            jpaAppDeploymentQueue.setProcess(processDAO.get(appDeploymentQueue.getProcessId()));
+            jpaAppDeploymentQueue.setAppDomain(appDeploymentQueue.getAppDomain());
+            jpaAppDeploymentQueue.setAppName(appDeploymentQueue.getAppName());
+            jpaAppDeploymentQueue.setUsers(usersDAO.get(principal.getName()));
+            Long adqId=appDeploymentQueueDAO.insert(jpaAppDeploymentQueue);
+            LOGGER.info("app deployment queue Id is "+adqId);
+             returnedAppDeploymentQueue.setProcessId(appDeploymentQueue.getProcessId());
+             returnedAppDeploymentQueue.setAppName(appDeploymentQueue.getAppName());
+             returnedAppDeploymentQueue.setAppDomain(appDeploymentQueue.getAppDomain());
+             returnedAppDeploymentQueue.setUsername(principal.getName());
+             returnedAppDeploymentQueue.setAppDeploymentQueueId(adqId);
+             returnedAppDeploymentQueue.setAppDeploymentStatusId((short) 0);
+             restWrapper = new RestWrapper(returnedAppDeploymentQueue, RestWrapper.OK);
+        } catch (MetadataException e) {
+            LOGGER.error(e);
+            restWrapper = new RestWrapper(e.getMessage(), RestWrapper.ERROR);
+        }
+        return restWrapper;
+    }
+
+
+    @RequestMapping(value = "/merge/{id}", method = RequestMethod.POST)
+    @ResponseBody
+    public
+    RestWrapper merge(@PathVariable("id") Long queueId, Principal principal) {
+        RestWrapper restWrapper = null;
+        LOGGER.info("queue id is "+queueId);
+        LOGGER.info("user name is "+principal.getName());
+        AppDeploymentQueue returnedAppDeploymentQueue=new AppDeploymentQueue();
+        try{
+            String temp;
+            BufferedReader br = null;
+            String jsonfile="";
+            String homeDir = System.getProperty("user.home");
+            LOGGER.info("home directory" + homeDir);
+            br = new BufferedReader(new FileReader(homeDir+"/bdreappstore/store.json"));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "/bdreappstore/store.json" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1192)